SpeechResponder/ConfigurationWindow.xaml.cs

```diff
@@ -7,379 +7,404 @@
+		}
 
-        public ConfigurationWindow()
-        {
-            InitializeComponent();
-            DataContext = this;
+		public ConfigurationWindow()
+		{
+			InitializeComponent();
+			DataContext = this;
 
-            ObservableCollection<Personality> personalities = new ObservableCollection<Personality>
-            {
-                // Add our default personality
-                Personality.Default()
-            };
-            // Add local personalities
-            foreach (Personality personality in Personality.AllFromDirectory())
-            {
-                if (personality != null)
-                {
-                    personalities.Add(personality);
-                }
-            }
-            Personalities = personalities;
+			ObservableCollection<Personality> personalities = new ObservableCollection<Personality>
+			{
+				// Add our default personality
+				Personality.Default()
+			};
+			// Add local personalities
+			foreach (Personality personality in Personality.AllFromDirectory())
+			{
+				if (personality != null)
+				{
+					personalities.Add(personality);
+				}
+			}
+			Personalities = personalities;
 
-            SpeechResponderConfiguration configuration = SpeechResponderConfiguration.FromFile();
-            subtitlesCheckbox.IsChecked = configuration.Subtitles;
-            subtitlesOnlyCheckbox.IsChecked = configuration.SubtitlesOnly;
+			SpeechResponderConfiguration configuration = SpeechResponderConfiguration.FromFile();
+			subtitlesCheckbox.IsChecked = configuration.Subtitles;
+			subtitlesOnlyCheckbox.IsChecked = configuration.SubtitlesOnly;
 
-            foreach (Personality personality in Personalities)
-            {
-                if (personality.Name == configuration.Personality)
-                {
-                    Personality = personality;
-                    break;
-                }
-            }
-        }
+			foreach (Personality personality in Personalities)
+			{
+				if (personality.Name == configuration.Personality)
+				{
+					Personality = personality;
+					break;
+				}
+			}
 
-        private void eddiScriptsEnabledUpdated(object sender, RoutedEventArgs e)
-        {
-            if (sender is CheckBox checkbox)
-            {
-                if (checkbox.IsLoaded)
-                {
-                    updateScriptsConfiguration();
-                }
-            }
-        }
+			Dispatcher.BeginInvoke(new Action(() =>
+			{
+				var recoveredScript = ScriptRecoveryService.GetRecoveredScript();
+				if (recoveredScript != null)
+				{
+					var messageBoxResult = MessageBox.Show(Properties.SpeechResponder.messagebox_recoveredScript,
+						Properties.SpeechResponder.messagebox_recoveredScript_title,
+						MessageBoxButton.YesNo, MessageBoxImage.Warning, MessageBoxResult.Yes,
+						MessageBoxOptions.DefaultDesktopOnly);
+					if (messageBoxResult == MessageBoxResult.Yes)
+					{
+						OpenEditScriptWindow(recoveredScript);
```

Comment:
The recoveredScript is passed to the `OpenEditScriptWindow()` method, but that method opens the edit script window referencing the version of the script saved to the personality rather than the version of the script recovered from the temporary file.